quesma/telemetry/phone_home.go

```diff
@@ -296,6 +296,63 @@ where active
 	return nil
 }
 
+func (a *agent) collectClickHouseTableSizes(ctx context.Context) (map[string]int64, error) {
+	tablesWithSizes, err := a.getTableSizes(a.ctx)
+	if err != nil {
+		logger.WarnWithCtx(ctx).Msgf("Error getting table sizes from clickhouse: %v", err)
+		return nil, err
+	}
+	return tablesWithSizes, nil
+}
+
+func (a *agent) getDbInfo() string {
+	dbUrl, dbName, dbUser := "", "default", "<no-user>"
+	if a.config.ClickHouse.User != "" {
+		dbUser = a.config.ClickHouse.User
+	}
+	if a.config.ClickHouse.Database != "" {
+		dbName = a.config.ClickHouse.Database
+	}
+	if a.config.ClickHouse.Url != nil {
+		dbUrl = a.config.ClickHouse.Url.String()
+	}
+	return fmt.Sprintf("%s@%s/%s", dbUser, dbUrl, dbName)
+}
+
+func (a *agent) getTableSizes(ctx context.Context) (map[string]int64, error) {
+	tableSizes := make(map[string]int64)
+	dbName := "default"
+	allTablesSize := int64(0)
+	if a.config.ClickHouse.Database != "" {
+		dbName = a.config.ClickHouse.Database
+	}
+	query := `SELECT table, sum(bytes_on_disk) AS total_size
+FROM system.parts
+WHERE active = 1 AND database = ?
+GROUP BY table
+ORDER BY total_size DESC;`
+
+	rows, err := a.clickHouseDb.QueryContext(ctx, query, dbName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute query: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var tableName string
+		var totalSize int64
+		if err := rows.Scan(&tableName, &totalSize); err != nil {
+			return nil, fmt.Errorf("failed to scan row: %w", err)
+		}
+		tableSizes[tableName] = totalSize
+		allTablesSize += totalSize
+	}
+	tableSizes["QUESMA_all_tables_in_bytes"] = allTablesSize
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("error iterating over rows: %w", err)
+	}
+	return tableSizes, nil
```

Comment:
We may have tons of tables and print them all; I'm not sure if this is desired. I would just print at most 100 of the biggest tables per size and total sum. I would also just print their size in MBs to prevent data leakage.